command/monitor/monitor.go

```diff
@@ -0,0 +1,172 @@
+package monitor
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+)
+
+// Monitor provides a mechanism to stream logs using go-hclog
+// InterceptLogger and SinkAdapter. It allows streaming of logs
+// at a different log level than what is set on the logger.
+type Monitor interface {
+	// Start returns a channel of log messages which are sent
+	// ever time a log message occurs
+	Start() <-chan []byte
+
+	// Stop de-registers the sink from the InterceptLogger
+	// and closes the log channels
+	Stop()
+}
+
+// monitor implements the Monitor interface
+type monitor struct {
+	// protects droppedCount and logCh
```

Comment:
Channels do their own locking internally, so I'm not sure you need a lock to protect logCh.  As to droppedCount, consider using an atomic instead.  There's nothing wrong with using a mutex, but I tend to prefer atomic if there's just one value involved.

Reply:
To be honest, I copied this file wholesale [from nomad](https://github.com/hashicorp/nomad/blob/master/command/agent/monitor/monitor.go).  It had the right public methods to make the rest of the pieces work, and was already merged as part of the PR to add `nomad monitor` so I didn't look closely at how it was implemented.  I don't know enough go at this point to spot its flaws anyway.

I can certainly look at addressing your comment, but I thought that context might be useful.

Reply:
I made these changes and everything still seems to work.  😁